tests/test_linalg.py

```diff
@@ -566,6 +566,54 @@ def test_inv_errors(self):
         assert_raises(inp.linalg.LinAlgError, inp.linalg.inv, a_dp)
 
 
+class TestMatrixPower:
+    @pytest.mark.parametrize("dtype", get_all_dtypes(no_bool=True))
+    @pytest.mark.parametrize(
+        "data, power",
+        [
+            (
+                numpy.block(
+                    [
+                        [numpy.eye(2), numpy.zeros((2, 2))],
+                        [numpy.zeros((2, 2)), numpy.eye(2) * 2],
+                    ]
+                ),
+                3,
+            ),  # Block-diagonal matrix
+            (numpy.eye(3, k=1) + numpy.eye(3), 3),  # Non-diagonal matrix
+            (
+                numpy.eye(3, k=1) + numpy.eye(3),
+                -3,
+            ),  # Inverse of non-diagonal matrix
+        ],
+    )
+    def test_matrix_power(self, data, power, dtype):
+        a = data.astype(dtype)
+        a_dp = inp.array(a)
+
+        result = inp.linalg.matrix_power(a_dp, power)
+        expected = numpy.linalg.matrix_power(a, power)
+
+        assert_dtype_allclose(result, expected)
+
+    def test_matrix_power_errors(self):
+        a_dp = inp.eye(4, dtype="float32")
+
+        # unsupported type `a`
+        a_np = inp.asnumpy(a_dp)
+        assert_raises(TypeError, inp.linalg.matrix_power, a_np, 2)
+
+        # unsupported type `power`
+        assert_raises(TypeError, inp.linalg.matrix_power, a_dp, 1.5)
+        assert_raises(TypeError, inp.linalg.matrix_power, a_dp, [2])
+
+        # not invertible
+        noninv = inp.array([[1, 0], [0, 0]])
+        assert_raises(
```

Comment:
public CI failed here on Windows, because no exception raised:
```
2024-03-16T08:56:08.6300244Z __________________ TestMatrixPower.test_matrix_power_errors ___________________
2024-03-16T08:56:08.6300884Z test_linalg.py:612: in test_matrix_power_errors
2024-03-16T08:56:08.6301324Z     assert_raises(
2024-03-16T08:56:08.6301895Z C:\Users\runneradmin\miniconda3\envs\test\lib\unittest\case.py:739: in assertRaises
2024-03-16T08:56:08.6302724Z     return context.handle('assertRaises', args, kwargs)
2024-03-16T08:56:08.6303420Z C:\Users\runneradmin\miniconda3\envs\test\lib\unittest\case.py:201: in handle
2024-03-16T08:56:08.6303918Z     callable_obj(*args, **kwargs)
2024-03-16T08:56:08.6304549Z C:\Users\runneradmin\miniconda3\envs\test\lib\unittest\case.py:223: in __exit__
2024-03-16T08:56:08.6305142Z     self._raiseFailure("{} not raised by {}".format(exc_name,
2024-03-16T08:56:08.6305759Z C:\Users\runneradmin\miniconda3\envs\test\lib\unittest\case.py:163: in _raiseFailure
2024-03-16T08:56:08.6306321Z     raise self.test_case.failureException(msg)
2024-03-16T08:56:08.6306761Z E   AssertionError: LinAlgError not raised by matrix_power
```
is it some known issue with `inv` on Windows and OpenCL CPU?